src/modelfree/shaping_wrappers.py

```diff
@@ -118,28 +138,35 @@ def predict(self, observation, state=None, mask=None, deterministic=False):
 
 
 def apply_reward_wrapper(single_env, shaping_params, agent_idx, scheduler):
-    anneal_frac = shaping_params.get('anneal_frac')
-    if anneal_frac is not None:
-        annealer = LinearAnnealer(1, 0, anneal_frac).get_value
-        scheduler.set_func('rew_shape', annealer)
-        rew_shape_func = scheduler.get_func('rew_shape')
+    if 'metric' in shaping_params:
+        rew_shape_annealer = ConditionalAnnealer.from_dict(shaping_params, get_logs=None)
+        scheduler.set_conditional('rew_shape')
     else:
-        # In this case, the different reward types are weighted differently
-        # but reward is not annealed over time.
-        rew_shape_func = None
-
+        anneal_frac = shaping_params.get('anneal_frac')
+        if anneal_frac is not None:
+            rew_shape_annealer = LinearAnnealer(1, 0, anneal_frac)
+        else:
+            # In this case, the different reward types are weighted differently
```

Comment:
Comment confusing. The sparse and dense reward are weighted the same -- both 1/2!